pctasks/server/pctasks/server/request.py

```diff
@@ -115,3 +141,37 @@ def request_access_key(self) -> Optional[str]:
         if not result:
             return None
         return result
+
+    @property
+    def custom_dimensions(self) -> Dict[str, Optional[str]]:
+        """Retrieves some common information from the request to log to App Insights"""
+
+        # 'request_id' header that can tie a traces entry to a request entry
+        # in App Insights. This is set by the API Management gateway.
+        # The header is in the format 'request_id: 1234.5678' where the first
+        # part is the request id that is logged in the traces entry.
+        request_id = _remove_after_dot(self._request.headers.get(HEADER_REQUEST_ID))
+
+        dimensions: Dict[str, Optional[str]] = {
+            "ref_id": self._request.headers.get(X_AZURE_REF),
+            "request_entity": get_request_entity(self._request),
+            "service": SERVICE_NAME,
+            HTTP_URL: str(self._request.url),
+            HTTP_METHOD: str(self._request.method),
+            HTTP_PATH: self.path,
+            DIMENSION_KEYS.REQUEST_ID: request_id,
```

Comment:
request_id is parsed from the "request-id" header that seems to be injected by APIM. Through trial and error I found that taking the value before the period would match up with the "id" in the "requests" table of AppInsights, allowing joining between the two logs.